x-pack/plugins/viz_editor/server/query/route.ts

```diff
@@ -24,16 +24,14 @@ export function route(server: Legacy.Server) {
     path: `${API_PREFIX}/search`,
     method: 'POST',
     async handler(req) {
-      const query = req.payload as Query;
+      const { query, indexpattern } = req.payload as { query: Query; indexpattern: string };
```

Comment:
To really support multiple data sources, won't we need to swap out the backends? This one seems really specific.

Also, any reason not to call it `indexPattern`?

Reply:
Yeah, this route is just for the `index_pattern` datasource, the others use their respective endpoint (or in case of CSV local data) - I would also pull this logic into the client and use `esaggs` and `esdocs` on the expression instead of `fancy_query`

Reply:
I'm totally fine with `indexPattern`